src/core/metrics.cc

```diff
@@ -310,6 +310,72 @@ instance_id_type shard() {
     return sstring("0");
 }
 
+void
+impl::set_metric_families_to_replicate(
+        std::unordered_map<seastar::sstring, int> metric_families_to_replicate) {
+    // Remove replicas for all metric families that are not
+    // present in the new map.
+    for (const auto& [name, destination]: _metric_families_to_replicate) {
+        if (!metric_families_to_replicate.contains(name)) {
+            remove_metric_replica_family(name, destination);
+        }
+    }
+
+    // Replicate the specified metric families, skipping the ones that
+    // already have replicas.
+    for (const auto& [name, destination]: metric_families_to_replicate) {
+        if (!_metric_families_to_replicate.contains(name)) {
+            replicate_metric_family(name, destination);
+        }
+    }
+
+    _metric_families_to_replicate = std::move(metric_families_to_replicate);
+}
+
+void impl::replicate_metric_family(const seastar::sstring& name, int destination) const {
+    const auto& entry = _value_map.find(name);
+
+    if (entry == _value_map.end()) {
+        return;
+    }
+
+    const auto& metric_family = entry->second;
+    for (const auto& [labels, metric_ptr]: metric_family) {
+        replicate_metric(metric_ptr, metric_family, destination);
```

Comment:
Destination is always the same here, `get_local_impl(destination)` could be lifted from `replicate_metric` to outside of the loop and passed down.

Reply:
I've lifted `get_local_impl` in `replicate_metric_family` and `remove_replicated_metric_family`. They were the worst offenders: queried for the `impl` for each metric in the family.